lib/acme/client.rb

```diff
@@ -127,9 +128,27 @@ def finalize(url:, csr:)
     Acme::Client::Resources::Order.new(self, **arguments)
   end
 
-  def certificate(url:)
+  def certificate(url:, preferred_chain: nil)
     response = download(url, format: :pem)
-    response.body
+    pem = response.body
+
+    return pem if preferred_chain.nil?
+
+    return pem if ChainIdentifier.new(pem).match_name?(preferred_chain)
+    # TODO: Remove Array() after decode_link_headers fix
+    #
+    #   FaradayMiddleware#decode_link_headers a single entry
+    #   per rel= but nothing prevent it in the spec.
+    alternative_urls = Array(response.headers['link']['alternate'])
```

Comment:
If you ask for a `preferred_chain`, and the CA does not respond with any `link`s, we need to protect against that here.

```suggestion
    alternative_urls = Array(response.headers.dig('link', 'alternate'))
```

Reply:
I'm not sure if that's explicitly addressed somewhere here in your TODO or not https://github.com/unixcharles/acme-client/pull/191

Reply:
Good call.